app/controllers/integrations/base_controller.rb

```diff
@@ -56,13 +58,14 @@ def create_new_release
     end
   end
 
-  def deploy_to_stages
-    stages = project.webhook_stages_for(branch, service_type, service_name)
+  def deploy_to_stages(stages)
     deploy_service = DeployService.new(user)
 
-    stages.all? do |stage|
+    return unless stages.all? do |stage|
       deploy_service.deploy!(stage, reference: commit).persisted?
```

Comment:
yep ... and since it uses `.all?` if you have 5 and the 2nd fails the only 1 stage got started ...